butane_core/src/many.rs

```diff
@@ -263,8 +264,20 @@ impl<T: DataObject> ManyOps<T> for Many<T> {
         .await?;
         self.new_values.clear();
         self.removed_values.clear();
-        // all_values is now out of date, so clear it
-        self.all_values = OnceLock::new();
+        // all_values is now out of date, so empty it
+        self.all_values = OnceLock::from(Vec::new());
+        Ok(())
+    }
+
+    async fn set(&mut self, conn: &impl ConnectionMethods, values: Vec<T>) -> Result<()> {
+        ManyOps::delete(self, conn).await?;
+
+        for value in &values {
+            self.add(value)?;
+        }
+        ManyOps::save(self, conn).await?;
```

Comment:
The perf of this can be vastly improved using bulk inserts, but IMO that can be done later, after the API of this new method has been finalised.

Also will probably require new DML abstractions for the bulk inserts.

If the Many object is already loaded, doing a set difference could also help avoiding performing unnecessary deletions and insertions, but I worry about that because the Many internals may be out of sync with the database if they have been modified by another connection to the database.

Reply:
Created https://github.com/Electron100/butane/issues/359 for the DML abstraction for bulk inserts.